packages/telemetry/src/flight-recorder.js

```diff
@@ -206,31 +181,212 @@ export const makeMemoryMappedCircularBuffer = async ({
+  stateDir = '/tmp',
+  circularBufferFilename,
+}) => {
+  const filename = circularBufferFilename || `${stateDir}/${DEFAULT_CBUF_FILE}`;
+
+  const newArenaSize = await initializeCircularBuffer(
+    filename,
+    circularBufferSize,
+  );
+
+  const file = await fsp.open(filename, 'r+');
+
+  const headerBuffer = Buffer.alloc(I_ARENA_START);
+
+  await file.read({
+    buffer: headerBuffer,
+    length: I_ARENA_START,
+    position: 0,
+  });
+  const header = new DataView(headerBuffer.buffer);
+
+  // Detect the arena size from the header, if not initialized.
+  const hdrArenaSize = header.getBigUint64(I_ARENA_SIZE);
+  const arenaSize = newArenaSize || hdrArenaSize;
+
+  const hdrMagic = header.getBigUint64(I_MAGIC);
+  SLOG_MAGIC === hdrMagic ||
+    Fail`${filename} is not a slog buffer; wanted magic ${SLOG_MAGIC}, got ${hdrMagic}`;
+  arenaSize === hdrArenaSize ||
+    Fail`${filename} arena size mismatch; wanted ${arenaSize}, got ${hdrArenaSize}`;
+
+  /** @type {(outbuf: Uint8Array, readStart: number, firstReadLength: number) => void} */
+  const readRecord = (outbuf, readStart, firstReadLength) => {
+    fs.readSync(file.fd, outbuf, {
+      length: firstReadLength,
+      position: Number(readStart) + I_ARENA_START,
+    });
+    if (firstReadLength < outbuf.byteLength) {
+      fs.readSync(file.fd, outbuf, {
+        offset: firstReadLength,
+        length: outbuf.byteLength - firstReadLength,
+        position: I_ARENA_START,
+      });
+    }
+  };
+
+  /**
+   * Writes to the file, offset by the header size
+   *
+   * @param {Uint8Array} record
+   * @param {number} firstWriteLength
+   * @param {bigint} circEnd
+   */
+  const writeRecord = (record, firstWriteLength, circEnd) => {
+    fs.writeSync(
+      file.fd,
+      record.subarray(0, firstWriteLength),
+      // TS saying options bag not available
+      undefined,
+      undefined,
+      Number(circEnd) + header.byteLength,
+    );
+    if (firstWriteLength < record.byteLength) {
+      // Write to the beginning of the arena.
+      fs.writeSync(
+        file.fd,
+        record.subarray(firstWriteLength, record.byteLength),
+        undefined,
+        undefined,
+        header.byteLength,
+      );
+    }
+  };
+
+  return finishCircularBuffer(arenaSize, header, readRecord, writeRecord);
+};
+
+/**
+ * @param {{
+ *   circularBufferSize?: number,
+ *   stateDir?: string,
+ *   circularBufferFilename?: string
+ * }} opts
+ */
+export const makeMemoryMappedCircularBuffer = async ({
+  circularBufferSize = DEFAULT_CBUF_SIZE,
+  stateDir = '/tmp',
+  circularBufferFilename,
+}) => {
+  const filename = circularBufferFilename || `${stateDir}/${DEFAULT_CBUF_FILE}`;
+
+  const newArenaSize = await initializeCircularBuffer(
+    filename,
+    circularBufferSize,
+  );
+
+  /**
+   * @type {Uint8Array}
+   * BufferFromFile mmap()s the file into the process address space.
+   */
+  const fileBuf = BufferFromFile(filename).Uint8Array();
```

Comment:
This function still uses BufferFromFile.  Shouldn't the need for Node 20 compatibility require removing `makeMemoryMappedCircularBuffer` and all references to it, including `packages/telemetry/src/frcat-entrypoint.js` and `packages/telemetry/test/test-flight-recorder.js`?

Reply:
yeah. I suppose I didn't explain well enough in the PR description: I thought it best to keep that one around until we're fully confident in the new one. It's trivial to remove later. 

this is why I made the issue "ref" not "close" the compatibility issue.